### PR TITLE
Upgrade storybook packages to close security vulnerabilities

### PR DESCRIPTION
**Pull request by alismx**
# DEVOPS PULL REQUEST

## Related Issue

- #6022 
- Uncontrolled resource consumption denial of service (ReDoS) vulnerability has been discovered in trim.
- A regular expression denial of service (ReDoS) vulnerability has been discovered in trim-newlines.
- [CWE-400](https://cwe.mitre.org/data/definitions/400.html)

## Changes Proposed

- Upgrade storybook to a version that doesn't pull in the vulnerable trim version
- Upgrade storybook packages to a version that doesn't pull in the vulnerable trim-newlines version
- Artifact Chomatic logs on failure
- Update Storybook, configurations, start/build scripts, import and Story usage to be compatible with the new versions

## Additional Info

- This is prep for turning on dependabot security PRs.

### Infrastructure
- [x] Consult the results of the `terraform-plan` job inside the "Terraform Checks" workflow run for this PR. Confirm that there are no unexpected changes!

### Security
- [x] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

### Cloud
- [ ] Oncall has been notified if this change is going in after-hours
- [ ] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### Documentation
- [ ] Any changes to the startup configuration have been documented in the README

### CODE REVIEW

**Review by emyl3 — COMMENTED** [review]
wow! thank you for working on this! 🙌  This will address this ticket as well 😸 https://github.com/CDCgov/prime-simplereport/issues/5899 :D

I am noticing when I start storybook up and open the local version (`http://localhost:6006/`) of it I see the following error:
Screenshot 2023-07-27 at 10 31 04

The "on your network" address works fine for me. 🤔

File: frontend/package.json [hunk 1]
```diff
@@ -77,8 +77,8 @@
     "compile-scss": "sass --load-path=./node_modules/uswds/dist/scss --style=compressed --no-source-map src/scss/App.scss:src/styles/App.css",
     "watch-css": "yarn compile-scss && sass --load-path=./node_modules/uswds/dist/scss --style=compressed --no-source-map -w src/scss/App.scss:src/styles/App.css",
     "create-storybook-public": "rm -rf ./storybook_public && cp -R ./public ./storybook_public && cp mockServiceWorker.js ./storybook_public",
-    "storybook": "yarn create-storybook-public && yarn watch-css & start-storybook -p 6006 -s storybook_public",
-    "build-storybook": "yarn create-storybook-public && yarn compile-scss && REACT_APP_BACKEND_URL=http://localhost:8080 build-storybook -s storybook_public",
+    "storybook": "yarn create-storybook-public && yarn watch-css & storybook dev -p 6006 -s storybook_public",
```
Comment (fzhao99) [inline frontend/package.json:80]:
```suggestion
    "storybook": "yarn create-storybook-public && yarn watch-css & storybook dev -p 6006 -s ../storybook_public",
```
@alismx I poked around for the issue Elisa raised and found [this GitHub issue](http://localhost:6006/?path=/docs/example-introduction--docs) and [this docs page](https://storybook.js.org/addons/msw-storybook-addon). I think the issue is that our yarn script here is pointing at the wrong filepath for the `storybook_public` directory where the MSW setup script was housed and so on startup, storybook can't find it. 

Changing this to the static dirs filepath to `../storybook_public` seems to have fixed the issue locally for me

Screenshot 2023-07-28 at 12 13 17 PM

Comment (alismx) [reply frontend/package.json:80]: Thanks so much! 🙌 

I thought something was wrong with msw or our worker file. I'll push your suggestion up asap.

**Review by mpbrown — APPROVED** [review]
Everything looks good locally! We'll be using Storybook throughout the process of revamping the SR design system so thank you for upgrading this!

**Review by fzhao99 — APPROVED** [review]

**Comment by alismx** [comment]
@johanna-skylight @fzhao99 @mpbrown It looks like this is saying we've got some UI review that needs to happen. Will at least two of you take a look? They look fine to me, but I want to double-check before merging this in.

**Review by johanna-skylight — APPROVED** [review]
Looks like Bob already approved the chromatic updates. Thanks for doing this Alis!! 💯

**Pull request merged**